src/components/webview/DiffWebviewPanel.ts

```diff
@@ -123,6 +141,36 @@ export class DiffWebviewPanel extends IntuitaWebviewPanel {
 		};
 	}
 
+	public async getViewDataForCase(
+		caseHash: ElementHash,
+	): Promise<Readonly<JobDiffViewProps>[]> {
+		const jobHashes = Array.from(
+			this.__caseManager.getJobHashes([
+				caseHash,
+			] as unknown as CaseHash[]),
```

Comment:
how can a job hash become a case hash? this looks like a serious issue to me

Reply:
`caseHash` is of type `ElementHash`.

```jsx
	public async getViewDataForCase(
		caseHash: ElementHash,
	): Promise<Readonly<JobDiffViewProps>[]> {
		const jobHashes = Array.from(
			this.__caseManager.getJobHashes([
				caseHash,
			] as unknown as CaseHash[]),
		);

		if (jobHashes.length === 0) {
			return [];
		}
		const viewDataArray = await Promise.all(
			jobHashes.map((jobHash) => this.getViewDataForJob(jobHash)),
		);
		return viewDataArray.filter(isNeitherNullNorUndefined);
	}
```

Reply:
I separated the original function that receives an argument of type `ElementHash | JobHash[]` into 2 functions:

```jsx
	public async getViewDataForCase(
		caseHash: ElementHash,
	): Promise<Readonly<JobDiffViewProps>[]> {
		const jobHashes = Array.from(
			this.__caseManager.getJobHashes([
				caseHash,
			] as unknown as CaseHash[]),
		);

		if (jobHashes.length === 0) {
			return [];
		}
		const viewDataArray = await Promise.all(
			jobHashes.map((jobHash) => this.getViewDataForJob(jobHash)),
		);
		return viewDataArray.filter(isNeitherNullNorUndefined);
	}

	public async getViewDataForJobsArray(
		jobHashes: JobHash[],
	): Promise<Readonly<JobDiffViewProps>[]> {
		if (jobHashes.length === 0) {
			return [];
		}
		const viewDataArray = await Promise.all(
			jobHashes.map((jobHash) => this.getViewDataForJob(jobHash)),
		);
		return viewDataArray.filter(isNeitherNullNorUndefined);
	}
```